src/hooks/useDividendCalculator.ts

```diff
@@ -0,0 +1,69 @@
+import { useHistoricalDividends } from '@/hooks/useHistoricalDividends';
+import { getDailyPortfolio } from '@/utils/portfolioCalculations/getDailyPortfolio';
+import { CurrentPortfolioItem } from '@/types/currentPortfolio';
+import { getSavedTransactions } from '@/utils/getSavedTransactions';
+import { useTransactionStore } from '@/stores/TransactionStore';
+import { getSortedTransactions } from '@/utils/portfolioCalculations/getSortedTransactions';
+
+export const useDividendCalculator = () => {
+  const savedTransactions = getSavedTransactions();
+  const existingTransactions = useTransactionStore(
+    (state) => state.transactions,
+  );
+  const transactions = [...existingTransactions, ...savedTransactions];
+  const startDate = getSortedTransactions(transactions)[0]?.transactionDate;
+
+  const endDate = new Date().toISOString().split('T')[0];
+
+  const dailyPortfolio = getDailyPortfolio({
+    existingTransactions,
+    savedTransactions,
+    startDate,
+    endDate,
+  });
+
+  const { data: dividendData, isLoading, error } = useHistoricalDividends();
+
+  const symbolDividendMap = dividendData?.reduce<Record<string, number>>(
+    (result, symbolData) => {
+      const { symbol, historical } = symbolData;
+
+      const totalDividendsForSymbol = historical.reduce(
```

Comment:
`symbolDividendMap` is not easily understandable. I would suggest to create a separate function for calculating `totalDividendsForSymbol`. And probably give a more meaningful naming for the used consts